python/mlx/nn/layers/convolution.py

```diff
@@ -34,6 +37,8 @@ def __init__(
         kernel_size: int,
         stride: int = 1,
         padding: int = 0,
+        dilation: int = 1,
+        groups: int = 1,
         bias: bool = True,
     ):
         super().__init__()
```

Comment:
Why the shape of the weight doesn't change? 
I think the shape should be (out_channel, kernel_size, in_channels/groups) and (in_channel%groups ==0).